tests/Backend/Snowflake/TypedTableWorkspacesLoadTest.php

```diff
@@ -0,0 +1,810 @@
+            $this->assertEquals('storage.tables.nonExistingColumns', $e->getStringCode());
+        }
+    }
+
+    public function testLoadIncrementalWithColumns(): void
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace();
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $tableId = $this->createTableLanguagesMoreColumns(['Id']);
+
+        // first load
+        $options = [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => 'languagesDetails',
+                    'whereColumn' => 'iso',
+                    'whereValues' => ['dd', 'xx'],
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'Name',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+//        $this->assertEquals(2, $backend->countRows("languagesDetails"));
+
+        // second load
+        $options = [
+            'input' => [
+                [
+                    'incremental' => true,
+                    'source' => $tableId,
+                    'destination' => 'languagesDetails',
+                    'whereColumn' => 'iso',
+                    'whereValues' => ['ff', 'xx'],
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'Name',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+        $this->assertEquals(5, $backend->countRows('languagesDetails'));
+    }
+
+    public function testSecondsFilter(): void
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace();
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $tableId = $this->createTableLanguages();
+        $importFile = __DIR__ . '/../../_data/languages.csv';
+        $originalFileLinesCount = (int) exec('wc -l <' . escapeshellarg($importFile));
```

Comment:
wut? to nejde jinak ?

Reply:
asi jo je to enom c&p :D

Reply:
dal sem tam count(iterator_to_array(CsvFile))